src/main/java/edu/cornell/kfs/coa/batch/service/impl/CopyLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+import org.kuali.kfs.krad.service.NoteService;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.krad.util.NoteType;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.cornell.kfs.coa.CuCOAKeyConstants;
+import edu.cornell.kfs.coa.batch.CopyLegacyAccountAttachmentsStep;
+import edu.cornell.kfs.coa.batch.CuCoaBatchConstants;
+import edu.cornell.kfs.coa.batch.CuCoaBatchParameterConstants;
+import edu.cornell.kfs.coa.batch.businessobject.LegacyAccountAttachment;
+import edu.cornell.kfs.coa.batch.dataaccess.CopyLegacyAccountAttachmentsDao;
+import edu.cornell.kfs.coa.batch.service.CopyLegacyAccountAttachmentsService;
+import edu.cornell.kfs.fp.batch.service.AccountingXmlDocumentDownloadAttachmentService;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentBackupLink;
+import edu.cornell.kfs.sys.service.WebServiceCredentialService;
+
+public class CopyLegacyAccountAttachmentsServiceImpl implements CopyLegacyAccountAttachmentsService {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final Pattern FILE_PATH_PATTERN = Pattern.compile("^(/\\w+)+$");
+
+    private CopyLegacyAccountAttachmentsDao copyLegacyAccountAttachmentsDao;
+    private AccountingXmlDocumentDownloadAttachmentService accountingXmlDocumentDownloadAttachmentService;
+    private WebServiceCredentialService webServiceCredentialService;
+    private AccountService accountService;
+    private PersonService personService;
+    private NoteService noteService;
+    private DataDictionaryService dataDictionaryService;
+    private ParameterService parameterService;
+    private ConfigurationService configurationService;
+
+    @Transactional
+    @Override
+    public boolean copyLegacyAccountAttachmentsToKfs() {
+        final int batchSize = getParameterValueAsInteger(CuCoaBatchParameterConstants.ATTACHMENT_BATCH_SIZE);
+        final int maxRetries = getParameterValueAsInteger(CuCoaBatchParameterConstants.MAX_DOWNLOAD_RETRIES);
+        final int maxFailuresAllowed = getParameterValueAsInteger(
+                CuCoaBatchParameterConstants.MAX_ATTACHMENT_FAILURES_PER_RUN);
+        final List<LegacyAccountAttachment> attachmentsToCopy = copyLegacyAccountAttachmentsDao
+                .getLegacyAccountAttachmentsToCopy(batchSize, maxRetries);
+        if (attachmentsToCopy.size() == 0) {
+            LOG.info("copyLegacyAccountAttachmentsToKfs, There are no legacy attachments available for copying");
+            return true;
+        }
+
+        LOG.info("copyLegacyAccountAttachmentsToKfs, Copying {} legacy account attachments into KFS",
+                attachmentsToCopy.size());
+        final CopyLegacyAccountAttachmentsService proxiedService = SpringContext.getBean(
```

Comment:
Why do we need a new instance of this service from SpringContext?  Can't we use the current instance of the service?

Reply:
Based on your prior review, I updated the code so that each legacy account attachment will be processed in its own transaction. Since I placed the transaction-annotated method on the same service class, the method has to be invoked on the transaction-proxied service so that the new transaction will be created and committed/rolled back accordingly.

Reply:
I'll add a method that wraps the `SpringContext` call and clarifies its intent.